src/components/Local.vue

```diff
@@ -73,80 +262,183 @@ export default {
+        this.reorder();
         this.newClip = "";
-        this.manage_cookie();
+        this.manageCookie();
       } else {
-        this.emptyString = true;
+        this.clearSnacks();
+        this.emptyStr = true;
       }
     },
-    copy(id) {
+    copyClip(id) {
+      // method to copy to the clipboard if the permission is given
+
       navigator.clipboard
         .writeText(this.clips[id].title)
         .then(() => {
-          this.copied = true;
+          this.clearSnacks();
+          this.boolCopied = true;
         })
         .catch(() => {
-          this.notCopied = true;
+          this.clearSnacks();
+          this.notBoolCopied = true;
         });
     },
-    delete_clip(id) {
+    deleteClip(id) {
+      // deleting the clip
+
       this.clips.splice(id, 1);
-      this.reshuffel();
-      this.manage_cookie();
-      this.just_got_deleted = true;
+      this.reorder();
+      this.manageCookie();
+      this.clearSnacks();
+      this.justGotDeleted = true;
     },
-    reshuffel() {
-      // reshuffling the array for the exact indexes
-      this.clips.filter(clip => {
-        clip.id = this.clips.indexOf(clip);
-      });
+    favouriteClip(id) {
+      // adding to favourite
+
+      if (this.clips[id].fav === true) {
+        this.clips[id].fav = !this.clips[id].fav;
+        this.manageCookie();
+        this.clearSnacks();
+        this.removedFromFav = true;
+      } else {
+        this.clips[id].fav = !this.clips[id].fav;
+        this.manageCookie();
+        this.clearSnacks();
+        this.addedToFav = true;
+      }
     },
-    manage_cookie() {
-      //  managing the cookies
-      this.$cookies.set("clips", JSON.stringify(this.clips), Infinity);
+    archiveClip(id) {
+      // adding to favourite
+
+      this.clips[id].archived = !this.clips[id].archived;
+      this.manageCookie();
+      this.clearSnacks();
+      this.justGotArchived = true;
+    },
+    previewCopy() {
+      let toCopy =
+        this.newClip.length > 0 ? this.newClip : "Start typing to see preview.";
+      navigator.clipboard
+        .writeText(toCopy)
+        .then(() => {
+          this.clearSnacks();
+          this.boolCopied = true;
+        })
+        .catch(() => {
+          this.clearSnacks();
+          this.notBoolCopied = true;
+        });
+    },
+    previewAddToFav() {
+      if (this.previewClip.fav === true) {
+        this.previewClip.fav = !this.previewClip.fav;
+        this.clearSnacks();
+        this.removedFromFav = true;
+      } else {
+        this.previewClip.fav = !this.previewClip.fav;
+        this.clearSnacks();
+        this.addedToFav = true;
+      }
+    },
+    clearSnacks() {
+      this.emptyStr = false;
+      this.boolCopied = false;
+      this.notBoolCopied = false;
+      this.justGotDeleted = false;
+      this.showArchived = false;
+      this.justGotArchived = false;
+      this.addedToFav = false;
+      this.removedFromFav = false;
```

Comment:
This function clears all the notifications that are already there and thus we can first call this functions and clear all notifications and after that, we can set value of specific notification var to true so that only that notification is left in the queue for example in the gif below, the existing notification doesn't disappear for 2 sec, but now, all the older notifications vanish and new comes up

![output](https://user-images.githubusercontent.com/45892659/82743734-bca3bf00-9d6f-11ea-9ea4-ccb02d5ebcea.gif)